source/core/slang-string-util.cpp

```diff
@@ -199,72 +199,95 @@ ComPtr<ISlangBlob> StringUtil::createStringBlob(const String& string)
     return (fromChar == toChar || string.indexOf(fromChar) == Index(-1)) ? string : calcCharReplaced(string.getUnownedSlice(), fromChar, toChar);
 }
 
-/* static */void StringUtil::calcLines(const UnownedStringSlice& textIn, List<UnownedStringSlice>& outLines)
+/* static */UnownedStringSlice StringUtil::extractLine(UnownedStringSlice& ioText)
```

Comment:
As an alternative to the special treatment of an empty string using `nullptr`, we could have this function have a signature like:

```c++
bool StringUtil::extractLine(UnownedStringSlice& ioText, UnownedStringSlice& outLine);
```

Where the logic is:

* If `ioText` contains at least one line separator, set `outLine` to the part before the first separator, set `ioText` to the part after the separator, and return `true`.
* Otherwise, set `outLine` to `ioText`, set `ioText` to an empty string, and return false.

Reply:
This is nicer and will change signature. In order to keep LineParser iterator working I need an Iterator value that indicates 'end' - and so whilst this could be done by storing a bool or the like, I think the 'special' slice is ok.